EpikCord/__init__.py

```diff
@@ -1891,7 +1894,93 @@ def unload_section(self, section: Union[CommandsSection, EventsSection]):
 # class ClientGuildMember(Member):
 #     def __init__(self, client: Client,data: dict):
 #         super().__init__(data)
+class VoiceWebsocketClient:
+    def __init__(self, client:Client):
+        self.client = client
+        self.HELLO = 8
+        self.IDENTIFY = 0
+        self.HEARTBEAT = 3
+        self.sequence = None
+
+    async def connect(self, guild_id:str, channel_id:str, **kwargs):
+        self.guild_id = guild_id
+        await self.client.send_json({
+            "op": 4,
+            "d" : {
+                "guild_id": guild_id,
+                "channel_id": channel_id,
+                "self_mute": kwargs.get("self_mute", False),
+                "self_deaf" : kwargs.get("self_deaf", False)
+            }
+        })
+        def figure_out_receive(data):
+            if data["t"] == "VOICE_SERVER_UPDATE":
+                return "VOICE_SERVER_UPDATE"
+            elif data["deaf"]:
+                return "VOICE_STATE_UPDATE"
+            else:
+                return ""
+        first_receive, second_receive = await asyncio.gather(await self.client.ws.receive(),await self.client.ws.receive())
+        receive1 = first_receive.json()
+        receive2 = second_receive.json()
+        if figure_out_receive(receive1) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive1
+        elif figure_out_receive(receive2) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive2
+        
+        if figure_out_receive(receive1) == "VOICE_STATE_UPDATE":
+            self.session_id=receive1["session_id"]
+        elif figure_out_receive(receive2) == "VOICE_STATE_UPDATE":
+            self.session_id=receive2["session_id"]
+
+        print(self.session_id)
+        print(voice_server_update)
+        self.token = voice_server_update["token"]
+        self.endpoint = voice_server_update["endpoint"]
+        
+        self.ws = await self.client.http.ws_connect("wss://"+ self.endpoint + "/?v=4")
+        await self.identify()
+        ready_resp = await self.ws.receive()
+        ready_parsed_data = await ready_resp.json()["d"]
```

Comment:
Please, just make it handle events like the WebsocketClient.

Reply:
As in, it uses the `async for event in self.ws`, and it makes of EventHandler.

Reply:
> Please, just make it handle events like the WebsocketClient.

in the new system, i registered the event to the WebsocketClient. More on discord chat